pkg/operator/operator.go

```diff
@@ -456,12 +456,24 @@ func (optr *Operator) maoConfigFromInfrastructure() (*OperatorConfig, error) {
 		return nil, err
 	}
 
-	// in case the MHC controller is disabled, leave its image empty
-	mhcImage := machineAPIOperatorImage
 	featureGates, err := optr.featureGateAccessor.CurrentFeatureGates()
 	if err != nil {
 		return nil, err
 	}
+
+	// Given we only support the MachineAPIMigration featuregate being passed by
+	// flags, it shold be the only thing populated our map (and therefore passed
+	// as args)
+	features := map[string]bool{
+		string(apifeatures.FeatureGateMachineAPIMigration): featureGates.Enabled(apifeatures.FeatureGateMachineAPIMigration),
+	}
+
+	if featureGates.Enabled(apifeatures.FeatureGateMachineAPIMigration) {
```

Comment:
Should this read from `features`? Would it help identify bugs in the construction of `features` if we read from it, or would it mask them? 

What's better, log line but machineset missing the arg, or both missing?

Reply:
I think it would be easier to debug both missing? So let's have the log line read from features.